test/src/main/java/org/jvnet/hudson/test/HudsonTestCase.java

```diff
@@ -1845,6 +1844,7 @@ public Page goTo(String relative, String expectedContentType) throws IOException
             Page p;
             try {
                 p = super.getPage(getContextPath() + relative);
+                WebClientUtil.waitForJSExec(this);
```

Comment:
Yep, that would probably work. In fact there are calls to `waitForJSExec` in a number of other places too that could probably be removed because I added calls in a number of places in `HtmlElementUtil` etc. I think I might just leave these for now though. It's a small thing and my head is wrecked enough with it :smile: